packages/components/src/list/experimental-list.tsx

```diff
@@ -0,0 +1,50 @@
+/**
+ * External dependencies
+ */
+import { Children, cloneElement, isValidElement } from '@wordpress/element';
+import { CSSTransition, TransitionGroup } from 'react-transition-group';
+
+type ListProps = {
+	component?: string;
+} & React.HTMLAttributes< HTMLElement >;
+
+export const ExperimentalList: React.FC< ListProps > = ( {
+	children,
+	component = 'nav',
+	// Allow passing any other property overrides that are legal on an HTML element
+	...otherProps
+} ) => {
+	return (
+		<TransitionGroup
+			component={ component }
+			className="woocommerce-list"
+			{ ...otherProps }
+		>
+			{ /* Wrapping all children in a CSS Transition means no invalid props are passed to children and that anything can be animated. */ }
```

Comment:
Unfortunately *any* direct child of TransitionGroup gets a bunch of props injected to it regardless of element type. react-transition-group is unfortunately not very actively maintained, so I've worked around this by wrapping all children in a CSSTransition, otherwise we get a lot of warnings that we're passing illegal props into the child components.

It means it is possible to animate any child of the list in the same way you can animate a normal list item, I'm not sure thats useful, but until we can move away from react-transition-group I don't see a better way to do this. Maybe we could create a dummy component that swallows the redundant props? I'm open to ideas.